bigframes/dtypes.py

```diff
@@ -205,93 +205,100 @@ class SimpleDtypeInfo:
 
 
 ## dtype predicates - use these to maintain consistency
-def is_datetime_like(type: ExpressionType) -> bool:
-    return type in (DATETIME_DTYPE, TIMESTAMP_DTYPE)
+def is_datetime_like(type_: ExpressionType) -> bool:
```

Comment:
`type` is a built-in. Avoid conflicts by using `type_`.

> `single_trailing_underscore_`: used by convention to avoid conflicts with Python keyword

https://peps.python.org/pep-0008/#descriptive-naming-styles

In this case, `type` is a function, not a keyword, so not strictly needed, but `type` is a common enough function that it is best to avoid conflicts, IMO.